Doctrine/Phpcr/SitemapUrlInformationProvider.php

```diff
@@ -0,0 +1,161 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\SeoBundle\Doctrine\Phpcr;
+
+use Doctrine\ODM\PHPCR\DocumentManager;
+use Jackalope\Transport\Logging\LoggerInterface;
+use PHPCR\Query\QueryInterface;
+use Psr\Log\LoggerInterface as PsrLogger;
+use Symfony\Cmf\Bundle\CoreBundle\PublishWorkflow\PublishWorkflowChecker;
+use Symfony\Cmf\Bundle\SeoBundle\AlternateLocaleProviderInterface;
+use Symfony\Cmf\Bundle\SeoBundle\Extractor\ExtractorInterface;
+use Symfony\Cmf\Bundle\SeoBundle\Model\UrlInformation;
+use Symfony\Cmf\Bundle\SeoBundle\SeoPresentation;
+use Symfony\Cmf\Bundle\SeoBundle\Sitemap\UrlInformationProviderInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\SecurityContextInterface;
+
+/**
+ * Providers are able to create a list of UrlInformation for the sitemap creation.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@gmx.de>
+ */
+class SitemapUrlInformationProvider implements UrlInformationProviderInterface
+{
+    /**
+     * @var AlternateLocaleProviderInterface
+     */
+    protected $alternateLocaleProvider;
+
+    /**
+     * @var ExtractorInterface
+     */
+    protected $titleExtractor;
+
+    /**
+     * @var DocumentManager
+     */
+    private $manager;
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var SecurityContextInterface
+     */
+    private $publishWorkflowChecker;
+    /**
+     * @var string
+     */
+    private $defaultChanFrequency;
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+    /**
+     * @var SeoPresentation
+     */
+    private $seoPresentation;
+
+    /**
+     * @param DocumentManager $manager
+     * @param RouterInterface $router
+     * @param string $defaultChanFrequency
+     * @param PsrLogger $logger
+     * @param SeoPresentation $seoPresentation
+     * @param SecurityContextInterface $publishWorkflowChecker
+     */
+    public function __construct(
+        DocumentManager $manager,
+        RouterInterface $router,
+        $defaultChanFrequency,
+        PsrLogger $logger,
+        SeoPresentation $seoPresentation,
+        SecurityContextInterface $publishWorkflowChecker = null
+    ) {
+        $this->manager = $manager ;
+        $this->router = $router;
+        $this->defaultChanFrequency = $defaultChanFrequency;
+        $this->logger = $logger;
+        $this->seoPresentation = $seoPresentation;
+        $this->publishWorkflowChecker = $publishWorkflowChecker;
+    }
+
+    /**
+     * {@inheritDocs}
+     */
+    public function getUrlInformation()
+    {
+        $routeInformation = array();
+
+        // todo rewrite query when https://github.com/symfony-cmf/CoreBundle/issues/126 is ready
+        $contentDocuments = $this->manager->createQuery(
+            "SELECT * FROM [nt:unstructured] WHERE (visible_for_sitemap = true)",
+            QueryInterface::JCR_SQL2
+        )->execute();
```

Comment:
maybe we can just move this bit to a protected `fetchSitemapDocuments` method. then it will be very easy to extend the class and adjust the query to other needs.

in the long run we could have two separate classes, one to fetch and one to transform. transforming an orm entity would probably look exactly the same, so it does not really belong in here. but these changes can be made without BC breaks, its just refactoring and moving things out.

i would suggest we just move creating and executing the query into a separate protected method.